openml/datasets/dataset.py

```diff
@@ -891,7 +894,7 @@ def _to_dict(self) -> "OrderedDict[str, OrderedDict[str, str]]":
 
         return data_container
 
-    def _get_arff(self, format: str) -> Dict:  # type: ignore
+    def _get_arff(self, format: str) -> Dict[str, Union[arff.DENSE, arff.COO]]:
```

Comment:
I think the return type of this needs to be the same as the inner `decode_arff`.

Reply:
Okay the issue here is since our Dict returned from __get_arff_ is a big Union, in _parse_data_from_arff_ function, all dict loads are complaining with all the type options dict indices / values can get. I added a typing ignore to that function for now, will go back to it later.